extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -194,8 +197,9 @@ export class BookTocManager implements IBookTocManager {
 	async updateTOC(version: BookVersion, tocPath: string, findSection?: JupyterBookSection, addSection?: JupyterBookSection): Promise<void> {
 		const tocFile = await fs.readFile(tocPath, 'utf8');
 		this.tableofContents = yaml.safeLoad(tocFile);
-		if (!this.tocFiles.has(tocPath)) {
-			this.tocFiles.set(tocPath, tocFile);
+		const toc = yaml.safeLoad(tocFile);
+		if (!this._initialToc.has(tocPath)) {
+			this._initialToc.set(tocPath, toc);
```

Comment:
I assume we don't want to have the same object reference since we'll be updating tableOfContents.

But you should use deepClone instead of safeLoad for multiple reasons : 

1. There's a very small but not impossible chance that the file changed between calls to safeLoad and we end up with different TOCs
2. For perf reasons we should avoid calling safeLoad as much as possible

Reply:
@lucyzhang929, I use safeLoad to create a brand new object.
@Charles-Gagnon, I'll deep clone the object instead.